src/sentry/static/sentry/app/views/releases/detail/overview/releaseStatsRequest.tsx

```diff
@@ -104,20 +129,30 @@ class ReleaseStatsRequest extends React.Component<Props, State> {
     }));
 
     try {
+      if (yAxis === YAxis.SESSIONS) {
+        data = await this.fetchSessions();
+      }
+
+      if (yAxis === YAxis.USERS) {
+        data = await this.fetchUsers();
+      }
+
       if (yAxis === YAxis.CRASH_FREE) {
-        data = await this.fetchRateData();
-      } else if (
+        data = await this.fetchCrashFree();
+      }
+
+      if (yAxis === YAxis.SESSION_DURATION) {
+        data = await this.fetchSessionDuration();
+      }
+
+      if (
         yAxis === YAxis.EVENTS ||
         yAxis === YAxis.FAILED_TRANSACTIONS ||
         yAxis === YAxis.COUNT_DURATION ||
         yAxis === YAxis.COUNT_VITAL
       ) {
+        // this is used to get total counts for chart footer summary
         data = await this.fetchEventData();
-      } else {
-        // session duration uses same endpoint as sessions
-        data = await this.fetchCountData(
-          yAxis === YAxis.SESSION_DURATION ? YAxis.SESSIONS : yAxis
-        );
```

Comment:
Should you raise an error if you leave the try block and data is undefined/null?

Reply:
That should not ever happen but I added a log to sentry so that we get notified if it for some reason happens 👍